flutter_local_notifications/docs/android-setup.md

```diff
@@ -0,0 +1,209 @@
+
+Foreground services require additions to the manifest. First, you must request the `FOREGROUND_SERVICE` permission, along with special permissions for each service type. Then, you must register the plugin service as shown below, adding any service types you may need. The exact names for each service type can be found at the link above.
+
+```xml
+<manifest ...>
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE"/>
+  <!-- Repeat this line for each service type -->
+  <uses-permission android:name="android.permission.FOREGROUND_SERVICE_MY_SERVICE_TYPE"/>
+  <application ...>
+    <!-- Customize the foregroundServiceType line below to your needs -->
+    <service
+      android:name="com.dexterous.flutterlocalnotifications.ForegroundService"
+      android:exported="false"
+      android:stopWithTask="false"
+      android:foregroundServiceType="my_service_type1|my_service_type2" >
+    </service>
+  </application>
+</manifest>
+```
+
+## Custom notification icons and sounds
+
+Notification icons should be added as a [drawable resource](https://developer.android.com/guide/topics/resources/drawable-resource), just like app icons. By default, the app's own icon is `@mipmap/ic_launcher`, and any such value can be passed directly to the `AndroidNotificationDetails()` constructor. For more details on creating custom notification icons, [see the docs](https://developer.android.com/studio/write/create-app-icons#create-notification). Custom notification sounds should be added to the `res/raw` directory.
+
+Notifications may also make use of [large icons](https://developer.android.com/develop/ui/views/notifications/expanded#image-style), such as album art or message attachments. When calling `show()`, pass a [`largeIcon`](https://pub.dev/documentation/flutter_local_notifications/latest/flutter_local_notifications/AndroidNotificationDetails/largeIcon.html) to the `AndroidNotificationDetails()` constructor. If you use a `DrawableResourceAndroidBitmap`, that indicates an image file in `/res/drawable`. Otherwise, use `FilePathAndroidBitmap` to point to any file.
+
+### Code and asset shrinking
+
+Flutter enables [code shrinking](https://developer.android.com/build/shrink-code) to minimize the release size, but this may lead to issues with [GSON](https://github.com/google/gson), a Java dependency used by this package. To fix this, copy the contents of [this file](https://github.com/google/gson/blob/main/examples/android-proguard-example/proguard.cfg) to `android/app/proguard-rules.pro`. If you use custom resources like icons or sounds, be sure to follow [these instructions](https://developer.android.com/build/shrink-code#keep-resources) to prevent them from getting deleted during the build process. You should also add the following lines to your _module build file_: 
+
+```groovy
+buildTypes {
+  release {
+    minifyEnabled true
+    shrinkResources false
```

Comment:
Can you please revert the commit that added this? I saw your response but would rather not have any mention of it as leads developers to not fix their issue and would rather have them do that and follow standard processes. From memory, resource shrinking is enabled as well by default for release builds as well

Reply:
Understandable, I'm away from my computer until tomorrow but will do